docs/en/developers/04_Pull_requests.md

```diff
@@ -0,0 +1,71 @@
+
+The rest of this document explains specific points.
+
+## How to rebase your branch on `master`
+
+Rebasing a branch is useful to make sure your code is based on the last version of FreshRSS and there is no conflicts. You have two options to do that.
+
+If you have any doubt, please let us know and we'll help you! We all began with Git one day and it's not an easy thing to work with.
+
+### Rebasing
+
+It's the cleanest method because the Git history will be all linear and so easier to read and to navigate into. It might also be more difficult if you're not at ease with Git since conflicts are harder to resolve.
+
+Note you should never rebase a branch if someone else is working on it. Otherwise, since it rewrites the history, it can be a real mess to sort it out.
+
+To rebase a branch:
+
+```console
+$ git checkout master      # go on master branch
+$ git pull upstream master # pull the last version of master
+$ git checkout -           # go back to your branch
+$ git rebase master        # rebase your branch on master
+```
+
+If you feel confident, you can use `git rebase -i master` to rewrite your history and make it clearer.
+
+### Merging
+
+If you prefer, you can simply merge `master` into your own branch. Conflicts might be easier to resolve, but your Git history will be less readable. Don't worry, we'll take care of it before merging your PR back into `master`.
+
+To merge `master`:
+
+```console
+$ git checkout master      # go on master branch
+$ git pull upstream master # pull the last version of master
+$ git checkout -           # go back to your branch
+$ git merge master         # merge master into your branch
+```
+
+## How to write a Git commit message
+
+It is important to have proper commits messages in order to facilitate later debugging, and so to read the following advice. Commits messages should inform about the choices made in the past (the “why?”)
+
+The first line should start by a verb (e.g. "Add") and explain in few words the objective of the commit. It's usually less than 50 characters so it stays concise. You can consider this line as the subject of your commit.
+
+Then, let an empty line, and start to write the body. It's usually less than 72 characters, but you are pretty free in the tone of the message. The body is the place where you can give the context of your patch. For instance, you can explain what you were doing when you identified a bug, or the problem you had before your patch. Giving these information help other developers to understand why a specific choice was made, especially when a patch introduces a bug that is identified months later.
```

Comment:
```suggestion
Then, insert a blank line, and start to write the body. It's usually wrapped at 72 characters, but you are pretty free in the tone of the message. The body is the place where you can clarify the context of your patch. For instance, you can explain what you were doing when you identified a bug, or the problem you had before your patch. Providing this information helps other developers understand why a specific choice was made, especially when a patch introduces a bug that is identified months later.
```

I assume you mean wrapped, because that's a rather bizarre statement otherwise. For the record, I object to such a statement about line wrapping in this context. I can't fathom why that's a common recommendation.

Reply:
That's it :+1:

I think the initial recommendation comes from the fact that git patches were meant to be shared by emails (the recommendation for emails is up to 78 characters though).

I usually wrap all my texts (commits, Markdown, HTML, […]) to 80 characters because I'm used to split my screen in two, and so the text doesn't move when I resize the windows. It's also easier to find one's way through the text and it makes updates easier (especially with Vim). My last reason is I prefer to stick to the "official" recommendations when they exist because tooling might make use of these.

My editor automatically wraps the lines so it's not something that bothers me but I understand it would be boring otherwise. This is the reason I wrote "usually": it doesn't make it a rule but only a recommendation :)